scio-core/src/test/scala/com/spotify/scio/transforms/FileDownloadDoFnTest.scala

```diff
@@ -42,9 +47,23 @@ class FileDownloadDoFnTest extends PipelineSpec {
     val tmpDir = Files.createTempDirectory("filedofn-")
     val files = createFiles(tmpDir, 100)
     runWithContext { sc =>
-      val p = sc.parallelize(files.map(_.toUri)).flatMapFile(fn, 10, false)
-      p.keys should containInAnyOrder((1 to 100).map(_.toString))
-      p.values.distinct should forAll { f: Path => !Files.exists(f) }
+      // try to use a single bundle so we can check
+      // elements flushed in processElement as well as
+      // elements flushed in finishBundle
+      val p = sc
+        .parallelize(Seq(files.map(_.toUri).zipWithIndex))
+        .flatten
+        .timestampBy { case (_, i) => new Instant(i + 1) }
+        .keys
+        .flatMapFile(fn, 10, false)
+        .withTimestamp
+
+      val contentAndTimestamp = p.map { case ((i, _), ts) => (i, ts.getMillis) }
+      val paths =  p.map { case ((_, f), _) => f }.distinct
+
+      val expected = (1L to 100L).map(i => (i.toString, i))
+      contentAndTimestamp should containInAnyOrder(expected)
```

Comment:
Ran on the original code, got non expected item `(1, 10)` -> (1st element was flushed while processing the 10th)